consensus/core/src/test_dag_builder.rs

```diff
@@ -123,64 +129,107 @@ impl DagBuilder {
             .collect::<Vec<VerifiedBlock>>()
     }
 
-    // TODO: reuse logic from Linearizer.
-    pub(crate) fn get_sub_dag_and_commit(
```

Comment:
Replaced with `get_sub_dag_and_commits` to re use the `Linearizer` logic to linearise the sub dag. That allow us to seamlessly reuse any new logic that has to do with gc_round and future refactorings.